ntoskrnl/mm/pagefile.c

```diff
@@ -747,7 +732,27 @@ NtCreatePagingFile(
         return STATUS_FLOPPY_VOLUME;
     }
 
-    PagingFile = ExAllocatePoolWithTag(NonPagedPool, sizeof(*PagingFile), TAG_MM);
+    /*
+     * Missing validation steps TODO:
+     * (see https://www.geoffchappell.com/studies/windows/km/ntoskrnl/api/mm/modwrite/create.htm )
+     *
+     * - Verify that no file system driver or any filter driver has done file
+     *   I/O while opening the file.
+     *   Verify that nothing of the paging file is yet in memory. Specifically,
+     *   the file object must either have no SectionObjectPointer or the latter
+     *   must have neither a DataSectionObject nor an ImageSectionObject.
+     *   Otherwise, we should fail, returning STATUS_INCOMPATIBLE_FILE_MAP.
+     *
+     * - Inform all the applicable drivers to prepare for the possibility of
+     *   paging I/O. Much of the point to paging I/O is to resolve page faults.
+     *   Especially important is that drivers that handle paging I/O do not
```

Comment:
drivers that handle ->  drivers handling

Reply:
both are correct.